lib/interactor/context.rb

```diff
@@ -121,10 +121,22 @@ def failure?
     #
     # Raises Interactor::Failure initialized with the Interactor::Context.
     def fail!(context = {})
-      error_context = Context.new
-      context.each { |key, value| error_context[key.to_sym] = value }
-      error_context.failure = @failure = true
-      raise Failure, error_context
+      context.each { |key, value| self[key.to_sym] = value }
```

Comment:
So the original code 

1. creates a new and separate `error_context`
2. copies all the current context's values over to it
3. marks both this context and error context as `failed`

Why not follow the same pattern in terms of creating a new, separate `error_context` to modify? The code below modifies the current context directly, which should probably be preserved (or at minimum conform as closely as possible to the original design decision the authors made).

Reply:
The issue with doing this on a different context is that we still end up passing the original context back.  We could update the original context with the params and redact passwords from an error context that would prevent Honey Badger from receiving this info while allowing the methods on the interactor that are needed after a fail! call.